src/mono/browser/runtime/assets.ts

```diff
@@ -7,15 +7,18 @@ import cwraps from "./cwraps";
 import { mono_wasm_load_icu_data } from "./icu";
 import { Module, loaderHelpers, mono_assert, runtimeHelpers } from "./globals";
 import { mono_log_info, mono_log_debug, parseSymbolMapFile } from "./logging";
-import { mono_wasm_load_bytes_into_heap_persistent } from "./memory";
+import { mono_wasm_load_bytes_into_heap_persistent, mono_wasm_load_stream_into_heap_persistent } from "./memory";
 import { endMeasure, MeasuredBlock, startMeasure } from "./profiler";
 import { AssetEntry } from "./types";
 import { VoidPtr } from "./types/emscripten";
 import { setSegmentationRulesFromJson } from "./hybrid-globalization/grapheme-segmenter";
 
 // this need to be run only after onRuntimeInitialized event, when the memory is ready
-export function instantiate_asset (asset: AssetEntry, url: string, bytes: Uint8Array): void {
-    mono_log_debug(`Loaded:${asset.name} as ${asset.behavior} size ${bytes.length} from ${url}`);
+export async function instantiate_asset (asset: AssetEntry, url: string, bytesOrStream: Uint8Array | [ReadableStream, number]) : Promise<void> {
```

Comment:
`[ReadableStream, number]` is not great. It's also easy to confuse with `Uint8Array` which is array-like

Could we use `asset.knownLength` instead  and have `bytesOrStream: Uint8Array | ReadableStream` here ?

Reply:
Yeah now that asset.knownLength exists we can use that. It didn't exist before